### DISCUSSION TITLE
Prevent code-server from opening files on start up

### DISCUSSION THREAD

**Discussion by NicoPaz123** [coder/code-server#5950] [Q&A]
Is there a way to stop code-server from opening files that were opened on an earlier session? Where is session information and state stored?

**Comment by lg-nineoneninethreenine** [comment]
I don't think it is related to session storage and the like. Rather, I guess it has to do with your settings. Try searching "tab" in the Settings UI, I'm sure the setting is there for it.

**Reply by NicoPaz123** [reply]
Tried a lot of this but nothing's working. If I have a specific workspace and I open a file form any folder of that workspace and close the browser then when I again access the browser code-server still keeps that file open. Is there any setting command for it? cause "windows.restoreWindows":"none" works for vscode but not for code-server.

**Reply by lg-nineoneninethreenine** [reply]
Well, not any that I know of. @jsjoeio is this a bug or something?

**Reply by jsjoeio** [reply]
Hmm...I'm not familiar with that option but it could be a bug! If it's reproducible in Codespaces, then it's a bug upstream. Otherwise, it's a code-server bug.

**Reply by lg-nineoneninethreenine** [reply]
Well, who the heck can use Codespaces now? The thing is gone for me, saying I drained it permanently.

**Comment by jsjoeio** [comment]
Maybe you're looking for this CLI flag? 

```shell
   -e --ignore-last-opened               Ignore the last opened directory or workspace in favor of an empty window.
```

**Reply by NicoPaz123** [reply]
This also keeps the previously opened files as tabs...searched a lot on this but not able to get any solution. If you open a workspace and keep some files open there and close the browser there's no way to keep the workspace but close those opened files when you access the browser again....

**Reply by jsjoeio** [reply]
Ahhhh... Hmm 🤔 Does this reproduce in Codespaces?

**Reply by lg-nineoneninethreenine** [reply]
Can't tell. The thing is gone for me because I drained the billing. Anyone tried Gitpod?

**Reply by NicoPaz123** [reply]
Haven't tried Codespaces or Gitpod, but the "windows.restoreWindows":"none" works for vscode I have tried that but doesn't work for code-server

**Reply by NicoPaz123** [reply]
Tried it on GitPod, "windows.restoreWindows":"none" doesn't work there as well.

**Reply by jsjoeio** [reply]
Ahhh...if it doesn't work on GitPod then I doubt it will work on Codespaces. If you can reproduce there (or vscode.dev) then I'd open a bug upstream.

**Reply by lg-nineoneninethreenine** [reply]
That setting works on vscode.dev, considering Microsoft made it. Maybe you could try using Codespaces under the free budget given by GitHub @jsjoeio, mine is just cleared and barred from usage.
